torch_xla/csrc/tensor_methods.cpp

```diff
@@ -1876,7 +1876,7 @@ std::pair<XLATensorPtr, XLATensorPtr> XLATensor::nms(
 XLATensorPtr XLATensor::nonzero(const XLATensorPtr& input) {
   torch::lazy::NodePtr node =
       torch::lazy::MakeNode<NonZero>(input->GetIrValue());
-  return input->CreateFrom(torch::lazy::Value(node, 0), at::ScalarType::Long);
+  return input->CreateFrom(torch::lazy::Value(node, 0));
```

Comment:
I assume what this `CreateFrom` does is: `input->GetIrValue()`(line1878) contains the IR graph before we add the `NonZero` node; then `input->CreateFrom` insert the `NonZero` node to the existing IR graph. Is my understanding correct?

Reply:
`CreateFrom` will create a new XLATensor using the IR node provided but it will use the same logical_element_type and device from the input tensor. I used `Create` here instead.